### DISCUSSION TITLE
Нерабочие триггеры "Роль добавлена" и "Роль убрана"

### DISCUSSION THREAD

**Discussion by rovius** [LacunaHub/.github#7] [Общее]
Автоматизации с триггерами "Роль добавлена" и "Роль убрана" не реагируют ни на чьи действия с ролями, будь то слеш-команда, интерактивное сообщение или ручная выдача роли участнику. Не выполняются действия как бесплатные, так и включенные в Diamond. В частности, не отправляются сообщения в заданный канал при выдаче участнику роли [(ветка)](https://discord.com/channels/740586549145763960/1223908360152416288).

**Comment by danliyev** [comment]
Тут мы скорее всего никак исправить это не сможем, т.к. если пользователь не находился в кэше, то и данные события триггериться не будут. Эти события привязаны к событию `GUILD_MEMBER_UPDATE` и если пользователь не был в кэше, то оба объекта идентичны и сравнение ролей не происходит, соответственно, события не триггерятся. Думаю, добавлю пояснялку на самом сайте, либо выпилю эти события отовсюду, т.к. их срабатывание чаще всего не происходит
